hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/formats/v6/BlockRecordWriterV6.java

```diff
@@ -273,15 +273,14 @@ public void close(@NonNull final HashObject endRunningHash) {
             if (gzipOutputStream != null) gzipOutputStream.flush();
             fileOutputStream.flush();
 
+            closeSidecarFileWriter();
             writeFooter(endRunningHash);
 
             outputStream.close();
             bufferedOutputStream.close();
             if (gzipOutputStream != null) gzipOutputStream.close();
             fileOutputStream.close();
 
-            closeSidecarFileWriter();
```

Comment:
There should be some test that verifies this. Both at the unit level and at the hapi spec level to verify e2e.

Reply:
Created https://github.com/hashgraph/hedera-services/issues/12098 and will find a volunteer for it. 😇